src/runner_manager.py

```diff
@@ -365,6 +379,9 @@ def reconcile(self, quantity: int, resources: VirtualMachineResources) -> int:
             len(runner_states.unhealthy),
         )
 
+        if self.config.charm_state.is_metrics_logging_available:
+            runner_metrics.extract(flavor=self.app_name, ignore_runners=set(runner_states.healthy))
```

Comment:
Extract can raise many types of exception, shouldn't we manage them (some) here to avoid breaking the reconcile loop ?

Reply:
Same as above, the only expected (not explicitly caught) exceptiond would be `CorruptMetricDataError` and an `OSError` if any of the metric files could not be removed and I hesitate to catch them for the reasons mentioned above.